develop/tutorials/articles/192-item-selector/02-creating-custom-entites-for-the-item-selector.markdown

```diff
@@ -1,151 +1,131 @@
-the detail steps.
-
-Create the ItemSelectorView OSGI component:
-
-@Component
-public class TasksItemSelectorView
-	implements ItemSelectorView<TasksItemSelectorCriterion> {
-
-	public Class<TasksItemSelectorCriterion> getItemSelectorCriterionClass() {
-		return TasksItemSelectorCriterion.class;
-	}
-
-	public List<ItemSelectorReturnType> getSupportedItemSelectorReturnTypes() {
-		return _supportedItemSelectorReturnTypes;
-	}
-
-	public String getTitle(Locale locale) {
-		return "Tasks";
-	}
-
-	public boolean isShowSearch() {
-		return false;
-	}
-
-	public boolean isVisible(ThemeDisplay themeDisplay) {
-		return true;
-	}
-
-	public ServletContext getServletContext() {
-		return _servletContext;
-	}
-
-	public void renderHTML(
-			ServletRequest request, ServletResponse response,
-			MyTasksItemSelectorCriterion itemSelectorCriterion,
-			PortletURL portletURL, String itemSelectedEventName, boolean search)
-		throws IOException, ServletException {
-
-		// Pass the information that the view needs
-
-		ServletContext servletContext = getServletContext();
-
-		RequestDispatcher requestDispatcher =
-			servletContext.getRequestDispatcher("/tasks.jsp");
-
-		requestDispatcher.include(request, response);
-	}
-
-	@Reference(
-		target = "(osgi.web.symbolicname=com.liferay.tasks.web)", unbind = "-"
-	)
-	public void setServletContext(ServletContext servletContext) {
-		_servletContext = servletContext;
-	}
-
-	private ServletContext _servletContext;
-	private static final List<ItemSelectorReturnType>
-		_supportedItemSelectorReturnTypes = Collections.unmodifiableList(
-			ListUtil.fromArray(
-				new ItemSelectorReturnType[] {
-					new TasksItemSelectorReturnType(),
-				}));
-
-}
-
-In the tasks.jsp we need to include the markup for the different tasks and when 
-the user selects any task, we need to fire a javascript event with the name 
-itemSelectedEventName specified in the renderHTML containing a json object with 
-the information of the task id and the task creator user as follows
-
-Liferay.fire(
-	'<%= {eventName} %>',
-
-	{
-		data: {the-data-your-client-needs}
-	}
-
-);
-
+# Creating Custom Entities for the Item Selector [](id=creating-custom-entities-for-the-item-selector)
+
+So, your app requires users to select an item that the Item Selector isn't 
+configured for? No problem. You can create a new entity.
+
+This tutorial explains how to create a new entity for the Item Selector.
+
+Go ahead and get started.
+
+## Creating Item Selector Criterion [](id=creating-item-selector-criterion)
+
+First, you must create a new criterion for your entity. Follow these steps to 
+create an Item Selector criterion:
+
+1.  Create a class that extends `BaseItemSelectorCriterion`.
+
+    The example below creates a class called `TaskItemSelectorCriterion`:
+
+        public class TasksItemSelectorCriterion extends 
+        BaseItemSelectorCriterion {
+        
+        }
+        
+    This class acts as an identifier for Item Selector.
```

Comment:
Yeah. I know that this extends BaseItemSelectorCriterion which provides some methods, implemented from the ItemSelectorCriterion interface. I'll point that out and update the description. I think I got too caught up on the return type. thx!